app/controllers/claims/support/users_controller.rb

```diff
@@ -5,9 +5,48 @@ def index
     @users = @school.users
   end
 
+  def new
+    @user = Claims::User.new
+  end
+
+  def check
+    if user.valid?
+      @user = user.decorate
+    else
+      render :new
+    end
+  end
+
+  def create
+    if user.save
+      attach_user_to_school
+      send_invite_email_to_user
+      redirect_to claims_support_school_users_path(@school)
+      flash[:success] = "User added"
+    else
+      render :new
+    end
+  end
+
   private
 
   def set_school
     @school = Claims::School.find(params[:school_id])
   end
+
+  def user
+    @user ||= Claims::User.new(user_params)
+  end
+
+  def attach_user_to_school
+    Membership.create!(user_id: user.id, organisation_type: "School", organisation_id: @school.id)
+  end
+
+  def send_invite_email_to_user
+    NotifyMailer.send_school_invite_email(user, @school, sign_in_url).deliver_now
```

Comment:
```suggestion
    NotifyMailer.send_school_invite_email(user, @school, sign_in_url).deliver_later
```

This should probably be deliver_later, sorry for the bad example but the email should be in a background job to not break the request.